ui/dashboards/src/context/DatasourceStoreProvider.tsx

```diff
@@ -96,9 +99,18 @@ export function DatasourceStoreProvider(props: DatasourceStoreProviderProps) {
     async function getClient<Client>(selector: DatasourceSelector): Promise<Client> {
       const { kind } = selector;
       const [{ spec, proxyUrl }, plugin] = await Promise.all([findDatasource(selector), getPlugin('Datasource', kind)]);
-      return plugin.createClient(spec.plugin.spec, { proxyUrl }) as Client;
+
+      // allows extending client
+      const datasourceSpec = { ...spec.plugin.spec };
```

Comment:
Is this trying to do a copy to avoid issues with mutation from the code calling `onCreate` or something else? If the former, is this deep copying enough to do that?